netconf/transport_junos.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2018, Juniper Networks, Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/*
+Transport JunOS provides the ability to communicate with JunOS via local shell
+NETCONF interface (xml-mode netconf need-trailer).
+*/
+package netconf
+
+import (
+	"os/exec"
+)
+
+// TransportJunOS maintains the information necessary to communicate with JunOS
+// via local shell.
+type TransportJunOS struct {
+	transportBasicIO
+	cmd *exec.Cmd
+}
+
+// Close closes an existing local NETCONF session.
+func (t *TransportJunOS) Close() error {
+	if t.cmd != nil {
+		t.ReadWriteCloser.Close()
+	}
+	return nil
+}
+
+// Open creates a new local NETCONF session.
+func (t *TransportJunOS) Open() error {
+	var err error
+
+	t.cmd = exec.Command("xml-mode", "netconf", "need-trailer")
+
+	writer, err := t.cmd.StdinPipe()
```

Comment:
`w` instead of `writer`

Reply:
fixed with 775c1a6312e97236b01eff60168bf616c9e1c741